### PR TITLE
Upgrade setup-ocaml to version 2

### PR DESCRIPTION
**Pull request by leviroth**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -20,8 +20,12 @@ jobs:
...
 
+      - name: Ensure `apt` is up to date
+        run: sudo apt-get update
+        if: contains(matrix.os, 'ubuntu')
+
       - name: Use OCaml ${{ matrix.ocaml-version }}
-        uses: avsm/setup-ocaml@v1
+        uses: avsm/setup-ocaml@v2
         with:
           ocaml-version: ${{ matrix.ocaml-version }}
```
Comment (smorimoto) [inline .github/workflows/workflow.yml:26]: Use `ocaml-compiler` instead for v2.

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -20,8 +20,12 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Ensure `apt` is up to date
+        run: sudo apt-get update
+        if: contains(matrix.os, 'ubuntu')
```
Comment (smorimoto) [inline .github/workflows/workflow.yml:25]: Now we can get rid of them.

Comment (leviroth) [reply .github/workflows/workflow.yml:25]: Thanks!

**Comment by leviroth** [comment]
Thanks for those tips @smorimoto (and for your work on setup-ocaml more generally)!

While you're here, I was surprised not to see more of a speedup, and I was wondering if there's anything obvious I should change. E.g., does the big speedup in v2 require enabling dune-cache?

**Comment by smorimoto** [comment]
If I remember correctly, I think GitHub had a specification that doesn't really enable caching until you merge the PR.

**Pull request merged**